tools/ml-dataprep/bigquery.py

```diff
@@ -0,0 +1,104 @@
+
+        The fully qualified table name has the format [PROJECT_ID].[DATASET].[TABLE_NAME].
+
+        Args:
+            project: A string containing the GCP project id in which resides the dataset containg
+            the table.
+            dataset: A string containing the BigQuery dataset in which resides the table.
+            table: A string containing the table name.
+
+        Returns:
+            A string containig the fully qualified name of a BigQuery table.
+        """
+        return '{}.{}.{}'.format(project, dataset, table)
+
+    def run_query(self, query):
+        """Run a SQL query against the BigQuery engine.
+
+        Args:
+            query: A string containing the SQL query to run.
+
+        Returns:
+            A list containing the rows returned by the query execution.
+        """
+        print('Running the query')
+        print(query)
+        query_job = self._bq_client.query(query)
+        rows = query_job.result()
+        return rows
+
+    def export_table_as_csv(self, project, dataset, table, destination_uri):
+        """Export a BigQuery table to Google Cloud Storage.
+
+        Args:
+            project: A string containing Google Cloud project where the BigQuery dataset is located.
+            dataset: A string containing the BigQuery dataset in which the table is located.
+            table: A string containing the name of the table to export.
+            destination_uri: A string containing the Google Cloud Stiorage path where to export the
+            table.
+        """
+        print('Exporting {}:{}.{} to {}'.format(
+            project, dataset, table, destination_uri))
+        dataset_ref = self._bq_client.dataset(dataset, project=project)
+        table_ref = dataset_ref.table(table)
+        job_config = bigquery.job.ExtractJobConfig()
+        job_config.print_header = False
+        extract_job = self._bq_client.extract_table(table_ref,
+                                                    destination_uri,
+                                                    location='US',
+                                                    job_config=job_config)
+        extract_job.result()  # Waits for job to complete.
+
+    def delete_table(self, table):
+        """Delete a BigQuery table.
+
+        Args:
+            table: A string containing the fully qualified BigQuery name of the table to delete.
+        """
+        print('Deleting table {}'.format(table))
+        if table is not None:
+            self._bq_client.delete_table(table)
+
+    def count_lines_in_table(self, table):
+        """Count the total number of lines in a table.
+
+        Args:
+            table: A string containing the fully qualified BigQuery name of the table.
+        """
+        query = queries.QUERY_COUNT_TEMPLATE.format(table)
+        rows = self.run_query(query)
+        for row in rows:
+            lines = row.lines
+        return lines
```

Comment:
you can get row counts from table object itself without needing to run a query:

```python
dataset_ref = client.dataset(dataset_id)
table_ref = dataset_ref.table(table_id)
table = client.get_table(table_ref)  # API Request

# View table properties
print(table.num_rows)
```

https://cloud.google.com/bigquery/docs/tables#getting_information_about_tables

Reply:
Done in commit fbe7635e89b78d502c8353cdbdd0d2b70de156b4